site/app/models/GradeableComponent.php

```diff
@@ -87,8 +87,8 @@ public function __construct(Core $core, $details=array()) {
         $this->order = $details['gc_order'];
         if (isset($details['gcd_score']) && $details['gcd_score'] !== null) {
             $this->has_grade = true;
-            $this->grader = isset($details['gcd_grader']) ? $details['gcd_grader']: null;
-            $this->graded_version = $details['gcd_graded_version'];
+            $this->grader = $details['gcd_grader'];
+            $this->graded_version = isset($details['gcd_graded_version']) ? $details['gcd_graded_version']: -1;
```

Comment:
I'd prefer this to be null as that makes more sense than "-1" being "not graded or missing".